src/fen.rs

```diff
@@ -38,11 +38,10 @@
 //! ```
 //! use shakmaty::{fen::Fen, CastlingMode, Chess, Position};
 //!
-//! let fen: Fen = "r1bqkbnr/ppp2Qpp/2np4/4p3/2B1P3/8/PPPP1PPP/RNB1K1NR b KQkq - 0 4".parse()?;
-//!
-//! let pos: Chess = fen.into_position(CastlingMode::Standard)?;
+//! const FEN: &str =  "r1bqkbnr/ppp2Qpp/2np4/4p3/2B1P3/8/PPPP1PPP/RNB1K1NR b KQkq - 0 4";
+//! let fen: Fen = FEN.parse().unwrap();
```

Comment:
Let's not use `.unwrap()` in examples (https://rust-lang.github.io/api-guidelines/documentation.html#examples-use--not-try-not-unwrap-c-question-mark).

In some cases we could use a more concrete hint than `# Ok::<_, Box<dyn std::error::Error>>(())`. If error types are mixed, hidden lines can be used as in the link above, or to add ad-hoc error conversions.

Reply:
Ah, I mentioned in one of the commit messages that using `Ok::<_, Box<dyn std::error::Error>>` fails in `no_std` because `std` is not linked. We do have one some options, though.

* In the case when there is only one error type, I could hard-code the `Err` variant to be that type.
* Otherwise, when there are multiple error types (i.e. `dyn Error` is necessary), we disable doc-tests unless `std` is turned on. That would require changing `test.yml`.
* The last option is to use `unwrap`, as I did in my revision.

Any thoughts on these?

Reply:
Update: in commit [10e8914](https://github.com/niklasf/shakmaty/pull/61/commits/10e891437135576c089e90d7671c2666f5d87649), I used the `?` operator for examples with homogenous error types. Still not sure how to proceed with the examples that require `Box<dyn std::error::Error>`, though.

Reply:
If all else fails, we could use

```
struct CommonError;
impl From<FooError> for CommonError { ... }
impl From<BarError> for CommonError { ... }
```

in hidden lines.

Reply:
Ah, I have considered this at one point, but I dismissed it because it felt too hacky. Nevertheless, it is a solution that works.

This should be resolved by 7ce3558. At my fork, all tests on CI now pass even without the `unwrap`. 🎉